Mathlib/Logic/Function/FiberPartition.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2024 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.Data.Set.Basic
+/-!
+
+This file defines the type `f.Fibers` of fibers of a function `f : Y → Z`, and provides some API
+to work with and construct terms of this type.
+
+Note: this API is designed to be useful when defining the counit of the adjunction between
+the functor which takes a set to the condensed set corresponding to locally constant maps to that
+set, and the forgetful functor from the category of condensed sets to the category of sets
+(see PR #14027).
+-/
+
+variable {X Y Z : Type*} (f : Y → Z) (g : X → Y)
+
+namespace Function
+
+/-- The indexing set of the partition. -/
+def Fibers : Type _ := Set.range (fun (x : Set.range f) ↦ f ⁻¹' {x.val})
```

Comment:
This won't work with the new `variable` algorithm. Please copy `f` syntax-explicitly into the statement:
```suggestion
def Fibers (f : Y → Z) : Type _ := Set.range (fun (x : Set.range f) ↦ f ⁻¹' {x.val})
```